source/channel.c

```diff
@@ -154,28 +184,18 @@ static void s_cleanup_slot(struct aws_channel_slot *slot) {
     }
 }
 
-void aws_channel_clean_up(struct aws_channel *channel) {
-
-    struct aws_channel_slot *current = channel->first;
-
-    if (!current) {
-        channel->channel_state = AWS_CHANNEL_SHUT_DOWN;
-        return;
-    }
-
-    if (!current->handler) {
-        channel->channel_state = AWS_CHANNEL_SHUT_DOWN;
-    }
+void aws_channel_destroy(struct aws_channel *channel) {
```

Comment:
The documentation says that aws_channel_shutdown() must have been called. Should we not assert on this at the top of this function?

Reply:
I put back the old logic which checked this. I didn't 100% grok it before. I added a comment explaining things.